lib/src/revset.rs

```diff
@@ -86,10 +86,13 @@ fn resolve_full_commit_id(
     symbol: &str,
 ) -> Result<Option<Vec<CommitId>>, RevsetError> {
     if let Ok(binary_commit_id) = hex::decode(symbol) {
+        if repo.store().hash_length() != binary_commit_id.len() {
+            return Ok(None);
+        }
```

Comment:
This feels like it's related to the first commit. Should it be there instead?

Reply:
It looks like this is no longer necessary in order to pass tests, so I've removed it altogether.

Reply:
Just kidding, it is still necessary (to avoid throwing an error when speculating that the revset expression may be a commit ID).